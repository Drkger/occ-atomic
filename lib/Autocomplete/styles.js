import spacing from '../subatomic/spacing';

export default {
    autoComplete: {
        position: 'relative'
    },
    droplist: {
        position: 'absolute',
        zIndex: 2,
        top: `calc(100% - ${spacing.xTiny}px)`,
        left: 0,
        width: '100%'
    }
};
