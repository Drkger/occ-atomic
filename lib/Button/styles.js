import colors from '../subatomic/colors';
import fonts from '../subatomic/fonts';
import icons from '../subatomic/icons';

export default {
    btn: {
        alignItems:'flex-start',
        boxSizing:'border-box',
        display:'inline-block',
        marginBottom:'0',
        textAlign:'center',
        whiteSpace:'nowrap',
        verticalAlign:'middle',
        touchAction:'manipulation',
        cursor:'pointer',
        userSelect:'none',
        padding:'0 30px',
        height:'30px',
        lineHeight:'24px',
        fontFamily:fonts.body,
        fontSize:'14px',
        fontWeight:'600',
        position:'relative',
        transition:'0.3s all',
        maxWidth:'100%',
        borderRadius:'5px',
        textDecoration:'none',
        '&:focus': {
            outline:'none'
        },
        '&:hover': {
            textDecoration:'none'
        }
    },
    cont: {
        display:'flex',
        width:'100%',
        height:'100%',
        alignItems:'center',
        justifyContent:'center',
        pointerEvents:'none'
    },
    loadCont: {
        display:'flex',
        width:'100%',
        height:'100%',
        alignItems:'center',
        justifyContent:'center',
        position:'absolute',
        top:'0',
        left:'0',
        pointerEvents:'none'
    },
    // Themes
    main: {
        background:colors.pink,
        color:colors.white,
        border:`1px solid ${colors.pink}`,
        '&:hover': {
            background:colors.pinkH,
            color:colors.white,
            borderColor:colors.pinkH
        },
        '&:active': {
            background:colors.pinkA,
            color:colors.white,
            borderColor:colors.pinkA
        },
        '& i': {
            background: props => props.icon ? icons.base(icons[props.icon].icon([colors.white])) : ''
        },
        '&$loading': {
            background:colors.pinkA,
            borderColor:colors.pinkA
        },
        '&$disabled': {
            background:`${colors.grey1} !important`,
            color:`${colors.white} !important`,
            border:`1px solid ${colors.grey1} !important`
        }
    },
    secondary: {
        background:'transparent',
        color:colors.pink,
        border:`1px solid ${colors.pink}`,
        '&:hover': {
            background:colors.pinkH,
            color:colors.white,
            borderColor:colors.pinkH,
            '& i': {
                background: props => props.icon ? icons.base(icons[props.icon].icon([colors.white])) : ''
            }
        },
        '&:active': {
            background:colors.pinkA,
            color:colors.white,
            borderColor:colors.pinkA,
            '& i': {
                background: props => props.icon ? icons.base(icons[props.icon].icon([colors.white])) : ''
            }
        },
        '& i': {
            background: props => props.icon ? icons.base(icons[props.icon].icon([colors.pink])) : ''
        },
        '&$loading': {
            background:colors.pinkA,
            borderColor:colors.pinkA
        },
        '&$disabled': {
            background:`${colors.grey1} !important`,
            color:`${colors.white} !important`,
            border:`1px solid ${colors.grey1} !important`
        }
    },
    white: {
        background:'transparent',
        color:colors.white,
        border:`1px solid ${colors.white}`,
        '&:hover': {
            background:'rgba(255,255,255,0.2)',
            color:colors.white,
            borderColor:colors.white,
            '& i': {
                background: props => props.icon ? icons.base(icons[props.icon].icon([colors.white])) : ''
            }
        },
        '&:active': {
            background:'rgba(255,255,255,0.2)',
            color:colors.white,
            borderColor:colors.white,
            '& i': {
                background: props => props.icon ? icons.base(icons[props.icon].icon([colors.white])) : ''
            }
        },
        '& i': {
            background: props => props.icon ? icons.base(icons[props.icon].icon([colors.white])) : ''
        },
        '&$loading': {
            background:'rgba(255,255,255,0.2)',
            borderColor:colors.white
        },
        '&$disabled': {
            background:`transparent !important`,
            color:`rgba(255,255,255,0.6) !important`,
            border:`1px solid rgba(255,255,255,0.6) !important`,
            '& i': {
                background: props => props.icon ? icons.base(icons[props.icon].icon(['rgba(255,255,255,0.6)'])) : ''
            }
        }
    },
    flat: {
        background:'transparent',
        color:colors.pink,
        border:`1px solid transparent`,
        '&:hover': {
            background:colors.grey3,
            color:colors.pink,
            borderColor:colors.grey3,
        },
        '&:active': {
            background:colors.grey5,
            color:colors.pink,
            borderColor:colors.grey5,
        },
        '& i': {
            background: props => props.icon ? icons.base(icons[props.icon].icon([colors.pink])) : ''
        },
        '& $loadIcon': {
            background: icons.base(icons.loading.icon([colors.pink]))
        },
        '&$loading': {
            background:colors.white,
            borderColor:colors.white
        },
        '&$disabled': {
            background:'transparent !important',
            color:`${colors.grey1} !important`,
            border:`1px solid transparent !important`,
            '& i': {
                background: props => props.icon ? icons.base(icons[props.icon].icon([colors.grey1])) : ''
            },
            '& $loadIcon': {
                background: icons.base(icons.loading.icon([colors.grey1]))
            },
        }
    },
    blue: {
        background:colors.blue,
        color:colors.white,
        border:`1px solid ${colors.blue}`,
        '&:hover': {
            background:colors.blueH,
            color:colors.white,
            borderColor:colors.blueH,
        },
        '&:active': {
            background:colors.blueA,
            color:colors.white,
            borderColor:colors.blueA,
        },
        '& i': {
            background: props => props.icon ? icons.base(icons[props.icon].icon([colors.white])) : ''
        },
        '&$loading': {
            background:colors.blueA,
            borderColor:colors.blueA
        },
        '&$disabled': {
            background:`${colors.grey1} !important`,
            color:`${colors.white} !important`,
            border:`1px solid ${colors.grey1} !important`
        }
    },
    loading: {
        pointerEvents:'none',
        '& $cont': {
            visibility:'hidden'
        }
    },
    disabled: {
        background:colors.grey1,
        color:colors.white,
        border:`1px solid ${colors.grey1}`,
        cursor:'default',
        pointerEvents:'none',
        '& i': {
            background: props => props.icon ? icons.base(icons[props.icon].icon([colors.white])) : ''
        },
        '&:hover, &:active, &:focus': {
            background:colors.grey1,
            color:colors.white,
            border:`1px solid ${colors.grey1}`
        }
    },
    // Sizes
    md: {
        fontSize:'16px',
        height:'40px',
        lineHeight:'28px',
        '& i': {
            width:'28px',
            height:'28px'
        }
    },
    lg: {
        fontSize:'18px',
        height:'46px',
        lineHeight:'32px',
        '& i': {
            width:'32px',
            height:'32px'
        }
    },
    // Block
    block: {
        display:'block',
        width:'100%'
    },
    // Icon
    icon: {
        width:'24px',
        height:'24px',
        display:'inline-block',
        transition:'0.3s all'
    },
    '@keyframes rotate': {
        from: {transform: 'rotate(0deg)'},
        to: {transform: 'rotate(360deg)'}
    },
    loadIcon: {
        width:'24px',
        height:'24px',
        display:'inline-block',
        transition:'0.3s all',
        animation:'rotate 1s infinite linear',
        background: icons.base(icons.loading.icon())
    }
};