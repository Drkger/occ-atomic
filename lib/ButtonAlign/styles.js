export default {
    sm: {
        marginTop:'19px'
    },
    md: {
        marginTop:'9px'
    },
    lg: {
        marginTop:'3px'
    }
};