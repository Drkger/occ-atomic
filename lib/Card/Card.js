import React from 'react';
import PropTypes from 'prop-types';

/** Container component with a card style */
class Card extends React.Component {

    render() {
        const { classes, children, noPadding, shadow, className, style, id } = this.props;
        return (
            <div id={id} className={`${!noPadding ? classes.card : classes.cardNoPadding}${shadow ? ` ${classes[`shadow${shadow}`]}` : ''}${className ? ` ${className}` : ''}`} style={style}>
                {children}
            </div>
        );
    }
}

Card.defaultProps = {
    shadow: 3
};

Card.propTypes = {
    classes: PropTypes.object,
    children: PropTypes.node,
    /** Set the level of shadow of the card. Choose one between 0 and 5. */
    shadow: PropTypes.oneOf([
        0, 1, 2, 3, 4, 5
    ]),
    /** The card has by default a padding. Use this property if you need to remove it. */
    noPadding: PropTypes.bool,
    id: PropTypes.string,
    className: PropTypes.string,
    style: PropTypes.object
};

export default Card;
