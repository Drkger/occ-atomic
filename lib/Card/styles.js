import colors from '../subatomic/colors';
import fonts from '../subatomic/fonts';
import shadows from '../subatomic/shadows';

const styles = {
    card: {
        background:colors.white,
        borderRadius:'5px',
        padding:'30px 20px'
    },
    cardNoPadding: {
        extend: 'card',
        padding:'0'
    }
};

const pushShadows = function() {
    for (let i = 0; i<7; i++) {
        styles[`shadow${i}`] = {
            boxShadow: shadows[`lvl${i}`]
        };
    }
};

pushShadows();

export default styles;