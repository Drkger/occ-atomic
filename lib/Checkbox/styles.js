import colors from '../subatomic/colors';
import spacing from '../subatomic/spacing';
import icons from '../subatomic/icons';

export default {
    cont: {
        paddingTop:spacing.tiny,
        paddingBottom:spacing.tiny,
        boxSizing:'border-box',
        display:'flex',
        alignItems:'center',
        cursor:'pointer',
        outline:'0',
        '&:after': {
            content:'""',
            display:'table',
            clear:'both'
        },
        '&:hover $check:after': {
            background:icons.base(icons.check.icon([colors.grey200]))
        }
    },
    check: {
        width:spacing.base,
        height:spacing.base,
        position:'relative',
        '&:before': {
            content:'""',
            width:spacing.small,
            height:spacing.small,
            borderRadius:4,
            position:'absolute',
            top:'50%',
            left:'50%',
            transform:'translate(-50%, -50%)',
            border:`1px solid ${colors.grey200}`,
            background:colors.bgWhite
        },
        '&:after': {
            content:'""',
            width:18,
            height:18,
            marginTop:-1,
            position:'absolute',
            top:'50%',
            left:'50%',
            transform:'translate(-50%, -50%)',
            transition:'0.3s all',
            background:'transparent'
        }
    },
    active: {
        '& $check:after': {
            background:[icons.base(icons.check.icon([colors.prim])), '!important']
        }
    },
    disabled: {
        opacity:0.4,
        pointerEvents:'none'
    },
    label: {
        marginLeft:spacing.tiny,
        cursor:'pointer'
    }
};
