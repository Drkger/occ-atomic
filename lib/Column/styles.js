import grid from '../subatomic/grid';

const styles = {
    col: {
        boxSizing:'border-box',
        position:'relative',
        minHeight:'1px',
        paddingRight:'15px',
        paddingLeft:'15px'
    }
};

const pushGrid = function() {
    const sm = `@media (min-width: ${grid.sm}px)`;
    const md = `@media (min-width: ${grid.md}px)`;
    const lg = `@media (min-width: ${grid.lg}px)`;
    for (let i = 1; i <= 12; i++) {
        styles[`xs${i}`] = {
            float:'left',
            width:`${100/12*i}%`
        };
        styles[`offset-xs${i}`] = {
            marginLeft:`${100/12*i}%`
        };
        styles[`push-xs${i}`] = {
            left:`${100/12*i}%`
        };
        styles[`pull-xs${i}`] = {
            right:`${100/12*i}%`
        };
    };
    styles[sm] = {};
    styles[md] = {};
    styles[lg] = {};
    for(let i = 1; i <= 12; i++) {
        styles[sm][`sm${i}`] = {
            float:'left',
            width:`${100/12*i}%`
        };
        styles[sm][`offset-sm${i}`] = {
            marginLeft:`${100/12*i}%`
        };
        styles[sm][`push-sm${i}`] = {
            left:`${100/12*i}%`
        };
        styles[sm][`pull-sm${i}`] = {
            right:`${100/12*i}%`
        };
        styles[md][`md${i}`] = {
            float:'left',
            width:`${100/12*i}%`
        };
        styles[md][`offset-md${i}`] = {
            marginLeft:`${100/12*i}%`
        };
        styles[md][`push-md${i}`] = {
            left:`${100/12*i}%`
        };
        styles[md][`pull-md${i}`] = {
            right:`${100/12*i}%`
        };
        styles[lg][`lg${i}`] = {
            float:'left',
            width:`${100/12*i}%`
        };
        styles[lg][`offset-lg${i}`] = {
            marginLeft:`${100/12*i}%`
        };
        styles[lg][`push-lg${i}`] = {
            left:`${100/12*i}%`
        };
        styles[lg][`pull-lg${i}`] = {
            right:`${100/12*i}%`
        };
    }
};

pushGrid();

export default styles;