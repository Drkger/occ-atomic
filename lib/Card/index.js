import injectSheet from 'react-jss';
import Card from './Card';
import styles from './styles';

const StyledCard = injectSheet(styles)(Card);
// eslint-disable-next-line
const { classes, ...propTypes } = Card.propTypes;
StyledCard.propTypes = {
	...propTypes,
	...StyledCard.propTypes
};

export default StyledCard;
