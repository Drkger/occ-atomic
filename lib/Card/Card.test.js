import React from 'react';
import { shallow } from 'enzyme';
import Card from './Card';
import styles from './styles';
import CardJSS from './index';

const reduceClasses = (prev, curr) => Object.assign({}, prev, { [curr]: curr });

const classes = Object.keys(styles).reduce(reduceClasses, {});

describe("Card", () => {

    it('matches the snapshot', () => {
        const wrapper = shallow(
            <Card classes={classes} />
        );
        expect(wrapper).toMatchSnapshot();
    });

    it('has no padding', () => {
        const wrapper = shallow(
            <Card classes={classes} noPadding />
        );
        expect(wrapper.hasClass('cardNoPadding')).toBe(true);
    });

    it('has the right level of shadow', () => {
        const wrapper = shallow(
            <Card classes={classes} shadow={3} />
        );
        expect(wrapper.prop('className')).toEqual('card shadow3');
    });

});

describe("CardJSS", () => {

    it('matches the snapshot', () => {
        const wrapper = shallow(
            <CardJSS />
        );
        expect(wrapper).toMatchSnapshot();
    });

});

describe("Card styles", () => {

    it('matches the snapshot', () => {
        expect(styles).toMatchSnapshot();
    });

});