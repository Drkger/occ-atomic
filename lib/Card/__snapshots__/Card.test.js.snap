// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    classes={
      Object {
        "card": "card",
        "cardNoPadding": "cardNoPadding",
        "shadow0": "shadow0",
        "shadow1": "shadow1",
        "shadow2": "shadow2",
        "shadow3": "shadow3",
        "shadow4": "shadow4",
        "shadow5": "shadow5",
        "shadow6": "shadow6",
      }
    }
    shadow={3}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": undefined,
      "className": "card shadow3",
      "id": undefined,
      "style": undefined,
    },
    "ref": null,
    "rendered": null,
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": undefined,
        "className": "card shadow3",
        "id": undefined,
        "style": undefined,
      },
      "ref": null,
      "rendered": null,
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Card styles matches the snapshot 1`] = `
Object {
  "card": Object {
    "background": "#ffffff",
    "borderRadius": "5px",
    "padding": "30px 20px",
  },
  "cardNoPadding": Object {
    "extend": "card",
    "padding": "0",
  },
  "shadow0": Object {
    "boxShadow": undefined,
  },
  "shadow1": Object {
    "boxShadow": "0 1px 4px rgba(0, 0, 0, 0.05)",
  },
  "shadow2": Object {
    "boxShadow": "0 1px 8px rgba(0, 0, 0, 0.05)",
  },
  "shadow3": Object {
    "boxShadow": "0 1px 10px rgba(0, 0, 0, 0.1)",
  },
  "shadow4": Object {
    "boxShadow": "0 1px 14px rgba(0, 0, 0, 0.1)",
  },
  "shadow5": Object {
    "boxShadow": "0 1px 18px rgba(0, 0, 0, 0.1)",
  },
  "shadow6": Object {
    "boxShadow": "0 6px 32px rgba(0, 0, 0, 0.1)",
  },
}
`;

exports[`CardJSS matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Jss(Card)
    shadow={3}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "classes": Object {
        "card": "Card-card-0-1-1",
        "cardNoPadding": "Card-cardNoPadding-0-1-2",
        "shadow0": "Card-shadow0-0-1-3",
        "shadow1": "Card-shadow1-0-1-4",
        "shadow2": "Card-shadow2-0-1-5",
        "shadow3": "Card-shadow3-0-1-6",
        "shadow4": "Card-shadow4-0-1-7",
        "shadow5": "Card-shadow5-0-1-8",
        "shadow6": "Card-shadow6-0-1-9",
      },
      "shadow": 3,
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "classes": Object {
          "card": "Card-card-0-1-1",
          "cardNoPadding": "Card-cardNoPadding-0-1-2",
          "shadow0": "Card-shadow0-0-1-3",
          "shadow1": "Card-shadow1-0-1-4",
          "shadow2": "Card-shadow2-0-1-5",
          "shadow3": "Card-shadow3-0-1-6",
          "shadow4": "Card-shadow4-0-1-7",
          "shadow5": "Card-shadow5-0-1-8",
          "shadow6": "Card-shadow6-0-1-9",
        },
        "shadow": 3,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
